react/src/base/inputs/SprkSelect/SprkSelect.js

```diff
@@ -0,0 +1,197 @@
+      id,
+      variant,
+      hasBlankFirstOption,
+      onChange,
+      ariaDescribedBy,
+      choices,
+      ...other
+    } = this.props;
+    const { choiceItems, selectHugeHasValue } = this.state;
+
+    return (
+      <select
+        className={classNames('sprk-b-Select', additionalClasses, {
+          'sprk-b-Input--has-floating-label': selectHugeHasValue,
+          'sprk-b-Select--error': !isValid,
+        })}
+        id={id}
+        disabled={isDisabled}
+        aria-describedby={ariaDescribedBy}
+        onChange={(e) => {
+          this.handleChange(e, variant, onChange);
+        }}
+        data-analytics={analyticsString}
+        data-id={idString}
+        ref={this.selectRef}
+        {...other}
+      >
+        {variant === 'huge' && (
+          // eslint-disable-next-line max-len
+          // eslint-disable-next-line jsx-a11y/control-has-associated-label
+          <option
+            value=""
+            hidden={!hasBlankFirstOption}
+            disabled={!hasBlankFirstOption}
+          />
+        )}
+        {choiceItems.map(
+          ({ id: innerId, label: innerLabel, options, value, ...rest }) => {
+            if (options) {
+              return (
+                <optgroup label={innerLabel} key={innerId}>
+                  {options.map(({ value: optionValue, label: optionLabel }) => (
+                    <option key={optionValue} value={optionValue} {...rest}>
+                      {optionLabel}
+                    </option>
+                  ))}
+                </optgroup>
+              );
+            }
+            return (
+              <option value={value} key={value} {...rest}>
+                {innerLabel}
+              </option>
+            );
+          },
+        )}
+      </select>
+    );
+  }
+}
+
+SprkSelect.propTypes = {
+  /**
+   * A space-separated string of classes to
+   * add to the outermost container of the component.
+   */
+  additionalClasses: PropTypes.string,
+  /**
+   * Assigned to the `data-analytics` attribute serving
+   * as a unique selector for outside libraries
+   * to capture data.
+   */
+  analyticsString: PropTypes.string,
+  /**
+   * Expects an array of objects
+   * that describe the choices available to the user.
+   */
+  choices: PropTypes.arrayOf(
+    PropTypes.shape({
+      /**
+       * Text tied to the rendered input element.
+       */
+      label: PropTypes.string.isRequired,
+      /**
+       * Assigned to the `name` attribute
+       * of the rendered input element.
+       */
+      name: PropTypes.string,
+      /**
+       * 	Assigned to the `value` attribute
+       *  of the rendered input element.
+       */
+      value: PropTypes.string,
+    }),
+  ).isRequired,
+  /**
+   * Will render the component in its disabled state.
+   */
+  isDisabled: PropTypes.bool,
+  /**
+   * Assigned to the `aria-describedby`
+   * attribute of the input used
+   * to create relationships between the
+   * input and error container.
```

Comment:
We should probably update these descriptions to not be specific to an 'error container' since it could also be helper text or another associated container.

Reply:
Yeah, lets do that